ext/date/php_date.stub.php

```diff
@@ -398,6 +398,12 @@ public static function __set_state(array $array) {}
 
 class DatePeriod implements IteratorAggregate
```

Comment:
I think we should decide what to do with the overloaded constructor:
- do nothing so that the overloaded signature remains
- deprecate it in 8.x and remove it in 9.0 altogether
- choose one "canonical" signature, deprecate the others in 8.x, and get rid of them in 9.0

Depending on the outcome, we could only have 2 factory methods, and the 3rd signature could be implemented by the constructor